.github/workflows/website-preview-cd.yml

```diff
@@ -45,11 +45,18 @@ jobs:
 
       - name: Restore Context
         run: |
-          cat action.env >> $GITHUB_ENV
+          PR_NUMBER=$(cat ./pr-number)
+          if ! [[ "$PR_NUMBER" =~ ^[0-9]+$ ]]; then
+            echo "Invalid PR number: $PR_NUMBER"
+            exit 1
+          fi
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
 
       - name: Set Deploy Name
         run: |
-          echo "DEPLOY_NAME=deploy-preview-${{ env.PR_NUMBER }}" >> $GITHUB_ENV
+          echo "DEPLOY_NAME=deploy-preview-${PR_NUMBER}" >> $GITHUB_ENV
+        env:
+          PR_NUMBER: ${{ env.PR_NUMBER }}
```

Comment:
This should not be necessary

Reply:
from @polarathene, directly using github action's context format syntax in scripts may cause command injection. Ref: https://securitylab.github.com/resources/github-actions-untrusted-input/#remediation

example vulnerable code:

```yaml
run:
  echo "DEPLOY_NAME=deploy-preview-${{ env.PR_NUMBER }}" >> $GITHUB_ENV
```

so i change this to always use env var at runtime instead.

Reply:
Thats correct but if `PR_NUMBER` is already defined as an env variable (`env.PR_NUMBER`) then you should be able to reference it from the bash script as you are doing securely (`${PR_NUMBER}`) without having to define a new env var since its already defined. Basically you are overwriting an env var with its current value so that can be removed